### PR TITLE
fix synk security scan on CI

### PR DESCRIPTION
**Pull request by joeyorlando**
# Which issue(s) this PR closes

Closes https://github.com/grafana/oncall/issues/4503
Closes https://github.com/grafana/oncall-private/issues/2876

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: .github/workflows/snyk-security-scan.yml [hunk 1]
```diff
@@ -23,8 +23,13 @@ jobs:
...
-        continue-on-error: true
-        run: snyk monitor --all-projects --severity-threshold=high
+      - name: snyk monitor
+        # https://docs.snyk.io/snyk-cli/commands/monitor
+        run: snyk monitor --all-projects --severity-threshold=high --exclude=dev,tools
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+      - name: snyk test
+        # https://docs.snyk.io/snyk-cli/commands/test
+        run: snyk test --all-projects --severity-threshold=high --exclude=dev,tools --fail-on=all
```
Comment (joeyorlando) [inline .github/workflows/snyk-security-scan.yml:33]: main change of this PR

File: grafana-plugin/package.json [hunk 2]
```diff
@@ -170,13 +170,14 @@
...
+    "stylelint": "^14.0.0",
     "stylelint-config-standard": "^22.0.0",
     "throttle-debounce": "^2.1.0",
     "tinycolor2": "^1.6.0",
     "tslib": "2.5.3"
   },
   "resolutions": {
-    "braces": "3.0.3"
+    "braces": "3.0.3",
+    "micromatch": "4.0.6"
```
Comment (joeyorlando) [inline grafana-plugin/package.json:181]: patches https://github.com/grafana/oncall-private/issues/2876 (and appeases the failing `synk test` 🙂)

File: .github/actions/setup-python/action.yml [hunk 3]
```diff
@@ -23,5 +23,5 @@ runs:
       if: ${{ inputs.install-dependencies == 'true' }}
       shell: bash
       run: |
-        pip install uv setuptools
+        pip install uv==0.2.37 setuptools==73.0.0
```
Comment (matiasb) [inline .github/actions/setup-python/action.yml:26]: Curious, why we need to pin here?

**Review by matiasb — APPROVED** [review]

File: .github/actions/setup-python/action.yml (hunk 3 above)
Comment (joeyorlando) [reply .github/actions/setup-python/action.yml:26]: good catch. I'll revert this, was just testing a few things.

`snyk` was complaining about [a CVE](https://security.snyk.io/vuln/SNYK-PYTHON-SETUPTOOLS-7448482) in `setuptools=68.0.0`. I'm not able to reproduce our environment installing this version though (`engine/requirements.txt` pins `73.0.0` and this GitHub Action, + our Dockerfile, also install `73.0.0` without the pin).. soo I permanently ignored this warning in snyk for now.

**Pull request merged**